pkg/campaign/handler.go

```diff
@@ -51,3 +53,23 @@ func (h *Handler) CreateCampaign(w http.ResponseWriter, r *http.Request) {
 
 	encode.JSON(w, campaign, http.StatusCreated)
 }
+
+// FindCampaignByID finds a campaign by :id
+func (h *Handler) FindCampaignByID(w http.ResponseWriter, r *http.Request) {
+	path := strings.Split(r.URL.String(), "/")
+
+	ok := len(path) == 4 && path[2] == "campaigns"
```

Comment:
Issue is that `mux.Vars` is missing after `r.WithContext`. Tried with `r.Clone` also but does not work.